src/components/TemplateSearchBar/TemplateSearchBar.tsx

```diff
@@ -5,12 +5,22 @@ import SearchIcon from '@mui/icons-material/Search';
 import { useTheme } from '@mui/material/styles';
 import { useTranslation } from '@i18n/client';
 
-interface TemplateSearchBarProps extends DictProps {}
+interface TemplateSearchBarProps extends DictProps {
+  onSearchTermChange: (searchTerm: string) => void;
+}
 
-export function TemplateSearchBar({ lang }: TemplateSearchBarProps) {
+export function TemplateSearchBar({
+  lang,
+  onSearchTermChange,
+}: TemplateSearchBarProps) {
   const { palette } = useTheme();
   const { t } = useTranslation(lang, 'components');
   const searchBarLabel = t('TemplateSearchBar.Label.text');
+
+  const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    onSearchTermChange(event.target.value);
+  };
+
   return (
     <TextField
```

Comment:
Currently the search bar is not centered on smaller screens but i tried this and it could be fixed with wrapping this TextField with
`<Box display="flex" justifyContent="center"></Box>`
And then setting width to 100%